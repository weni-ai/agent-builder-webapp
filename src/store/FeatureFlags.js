import { defineStore } from 'pinia';
import { computed, inject, watch } from 'vue';

import { useProjectStore } from './Project';

import { gbKey } from '../utils/Growthbook';
import env from '@/utils/env';

export const useFeatureFlagsStore = defineStore('FeatureFlags', () => {
  const growthbook = inject(gbKey);

  const currentProjectUuid = computed(() => useProjectStore().uuid || '');

  const getListAtEnv = (key) => {
    return env(key)?.split(',') || [];
  };

  const isProjectEnabledForFlag = (flagKey) => {
    const projectList = getListAtEnv(flagKey);
    return projectList.includes(currentProjectUuid.value);
  };

  const flags = computed(() => ({
    supervisorExport: isProjectEnabledForFlag('FF_SUPERVISOR_EXPORT'),
    newSupervisor: growthbook?.isOn('new_supervisor'),
  }));

  watch(
    currentProjectUuid,
    (newProjectUuid) => {
      if (newProjectUuid && growthbook) {
        growthbook.setAttributes({
          ...growthbook.getAttributes(),
          weni_project: newProjectUuid,
        });
      }
    },
    { immediate: true },
  );

  return {
    flags,
  };
});
