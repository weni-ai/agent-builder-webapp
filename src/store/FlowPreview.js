import { defineStore } from 'pinia';
import { ref } from 'vue';
import { get } from 'lodash';

import useFlowPreview from '@/composables/useFlowPreview';

export const useFlowPreviewStore = defineStore('flowPreview', () => {
  const { preview, previewInit } = useFlowPreview();

  const messages = ref([]);
  const messageInput = ref('');

  function addMessage(message) {
    messages.value.push(message);
  }

  function clearMessages() {
    messages.value = [];
  }

  function setMessageInput(text) {
    messageInput.value = text;
  }

  function replaceMessage(toReplace, newMessages) {
    const index = messages.value.indexOf(toReplace);
    if (index !== -1) {
      messages.value.splice(
        index,
        1,
        ...(Array.isArray(newMessages) ? newMessages : [newMessages]),
      );
    }
  }

  function removeMessage(message) {
    const index = messages.value.indexOf(message);
    if (index !== -1) {
      messages.value.splice(index, 1);
    }
  }

  function treatAnswerResponse(
    answer,
    data,
    { onBroadcast, onMediaUnavailable, onCancelled, fallbackMessage = '' },
  ) {
    if (data.type === 'broadcast') {
      handleBroadcastResponse(answer, data, fallbackMessage, onBroadcast);
    } else if (data.type === 'flowstart') {
      // Insert a flowstart message before the answer
      const answerIndex = messages.value.indexOf(answer);
      messages.value.splice(answerIndex, 0, {
        type: 'flowstart',
        name: data.name,
        question_uuid: null,
        feedback: {
          value: null,
          reason: null,
        },
      });
    } else if (data.type === 'media_and_location_unavailable') {
      answer.status = 'loaded';
      answer.type = data.type;

      if (onMediaUnavailable) onMediaUnavailable(answer);
    } else if (data.type === 'cancelled') {
      answer.status = 'loaded';
      removeMessage(answer);

      if (onCancelled) onCancelled();
    }
  }

  function handleBroadcastResponse(answer, data, fallbackMessage, onBroadcast) {
    answer.status = 'loaded';

    const message = get(data, 'message', fallbackMessage);
    const sources = get(data, 'fonts', []);

    if (Array.isArray(message) && message.length > 0) {
      answer.response = message[0];
      answer.sources = sources;

      createAdditionalMessages(message.slice(1), answer.question_uuid, sources);
    } else {
      answer.response = message;
      answer.sources = sources;
    }

    if (onBroadcast) onBroadcast(answer);
  }

  function createAdditionalMessages(items, questionUuid, sources) {
    items.forEach((item) => {
      const additionalMessage = {
        type: 'answer',
        status: 'loaded',
        response: item,
        question_uuid: questionUuid,
        sources,
        feedback: {
          value: null,
          reason: null,
        },
      };
      addMessage(additionalMessage);
    });
  }

  return {
    preview,
    messages,
    messageInput,

    previewInit,

    addMessage,
    clearMessages,
    setMessageInput,
    replaceMessage,
    removeMessage,
    treatAnswerResponse,
  };
});
