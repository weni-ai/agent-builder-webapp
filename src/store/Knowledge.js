
import { defineStore } from 'pinia';
import { reactive } from 'vue';

export const useKnowledgeStore = defineStore('Knowledge', () => {
  const contentText = reactive({
    uuid: null,
    current: '',
    old: '',
  });

  return {
    contentText,
  };
});
