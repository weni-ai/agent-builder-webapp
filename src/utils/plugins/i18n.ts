import { createI18n } from 'vue-i18n';

import en from '@/locales/en.json';
import ptbr from '@/locales/pt_br.json';
import es from '@/locales/es.json';

const languages = {
  en,
  'pt-br': ptbr,
  es,
};

const messages = Object.assign(languages);

const i18n = createI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
  warnHtmlInMessage: 'off',
});

export default i18n;
