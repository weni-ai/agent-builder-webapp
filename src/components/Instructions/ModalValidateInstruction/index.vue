<template>
  <UnnnicModalDialog
    data-testid="modal-validate-instruction-by-ai"
    :modelValue="modelValue"
    size="lg"
    showCloseIcon
    showActionsDivider
    :title="
      $t(
        'agent_builder.instructions.new_instruction.validate_instruction_by_ai.modal_title',
      )
    "
    :secondaryButtonProps="{
      text: $t(
        'agent_builder.instructions.new_instruction.validate_instruction_by_ai.cancel_button',
      ),
      onClick: close,
    }"
    :primaryButtonProps="{
      text: $t(
        'agent_builder.instructions.new_instruction.validate_instruction_by_ai.publish_button',
      ),
      disabled:
        instructionsStore.instructionSuggestedByAI.status === 'loading' ||
        newInstruction.trim() === '',
      loading: instructionsStore.newInstruction.status === 'loading',
      onClick: publishInstruction,
    }"
    @update:model-value="close"
  >
    <InstructionTextarea
      v-model="newInstruction"
      data-testid="modal-validate-instruction-textarea"
    />
  </UnnnicModalDialog>
</template>

<script setup lang="ts">
import { onMounted, ref } from 'vue';

import { useInstructionsStore } from '@/store/Instructions';

import InstructionTextarea from './InstructionTextarea.vue';

const emit = defineEmits(['update:modelValue']);

const instructionsStore = useInstructionsStore();

const modelValue = ref(false);

const newInstruction = ref(instructionsStore.newInstruction.text);

function close() {
  emit('update:modelValue', false);
}

function publishInstruction() {
  instructionsStore.newInstruction.text = newInstruction.value;
  instructionsStore.addInstruction();
}

onMounted(() => {
  const isInstructionNotSuggested =
    instructionsStore.instructionSuggestedByAI.data.instruction !==
    instructionsStore.newInstruction.text;

  if (isInstructionNotSuggested) {
    instructionsStore.getInstructionSuggestionByAI();
  }
});
</script>
