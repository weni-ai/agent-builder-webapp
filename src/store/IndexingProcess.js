import { defineStore } from 'pinia';
import { ref, watch, toValue } from 'vue';
import { useAlertStore } from './Alert';
import nexusaiAPI from '@/api/nexusaiAPI';
import i18n from '@/utils/plugins/i18n';

export const useIndexingProcessStore = defineStore('IndexingProcess', () => {
  const itemsBeingProcessed = ref([]);
  const alertStore = useAlertStore();

  function addItemBeingProcessed(type, itemUuid, item, contentBaseUuid) {
    const alreadyIn = itemsBeingProcessed.value.find(
      ({ uuid }) => uuid === itemUuid,
    );

    if (!alreadyIn) {
      itemsBeingProcessed.value.push({
        type,
        uuid: itemUuid,
        item,
        contentBaseUuid,
      });
    }
  }

  function removeFirstItemBeingProcessed() {
    return itemsBeingProcessed.value.shift();
  }

  function moveFirstItemBeingProcessedToTheEnd() {
    const firstItem = removeFirstItemBeingProcessed();
    itemsBeingProcessed.value.push(firstItem);
  }

  async function checkIfItemHasAlreadyBeenProcessed() {
    const { type, uuid, item, contentBaseUuid } =
      itemsBeingProcessed.value.at(0);

    await nexusaiAPI.intelligences.contentBases[type]
      .read({
        contentBaseUuid,
        uuid,
      })
      .then(({ data }) => {
        if (data.status === 'success') {
          removeFirstItemBeingProcessed();

          item.status = 'uploaded';
          item.file_name = data.file_name;
          item.created_at = data.created_at;

          if (type === 'files') {
            successMessage();
          }
        } else if (data.status === 'fail') {
          removeFirstItemBeingProcessed();

          item.status = data.status;
        } else {
          moveFirstItemBeingProcessedToTheEnd();
        }
      })
      .catch(() => {
        moveFirstItemBeingProcessedToTheEnd();
      });

    if (itemsBeingProcessed.value.length >= 1) {
      setTimeout(checkIfItemHasAlreadyBeenProcessed, 3000);
    }
  }

  watch(
    () => itemsBeingProcessed.value.length,
    (current, previous) => {
      const hasBeenAddedSomeItem = previous === 0 && current >= 1;

      if (hasBeenAddedSomeItem) {
        setTimeout(checkIfItemHasAlreadyBeenProcessed, 3000);
      }
    },
  );

  function successMessage() {
    alertStore.add({
      type: 'success',
      text: i18n.global.t(
        'content_bases.files.content_of_the_files_has_been_added',
      ),
    });
  }

  function useIndexingProcess(items, type, contentBaseUuid) {
    watch(
      () => items,
      (items) => {
        toValue(items)
          .filter(({ status }) => status === 'processing')
          .forEach((item) => {
            addItemBeingProcessed(type, item.uuid, item, contentBaseUuid);
          });
      },
      { deep: true },
    );
  }

  return {
    useIndexingProcess,
  };
});
