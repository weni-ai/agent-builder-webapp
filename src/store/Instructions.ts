import { reactive, ref, computed } from 'vue';
import { defineStore } from 'pinia';

import type { InstructionSuggestedByAI } from './types/Instructions.types';

import { useProjectStore } from './Project';
import { useAlertStore } from './Alert';

import nexusaiAPI from '@/api/nexusaiAPI';

import i18n from '@/utils/plugins/i18n';
import { moduleStorage } from '@/utils/storage';

function callAlert(type, alertText) {
  const alertStore = useAlertStore();
  alertStore.add({
    text: i18n.global.t(`agent_builder.instructions.${alertText}`),
    type,
  });
}

export const useInstructionsStore = defineStore('Instructions', () => {
  const projectUuid = computed(() => useProjectStore().uuid);

  const instructions = reactive({
    data: [],
    status: null,
  });

  const newInstruction = reactive({
    text: '',
    status: null,
  });

  const validateInstructionByAI = ref<boolean>(
    moduleStorage.getItem('validateInstructionByAI') ?? true,
  );

  const instructionSuggestedByAI = reactive<InstructionSuggestedByAI>({
    data: {
      instruction: '',
      classification: [],
      suggestion: '',
    },
    suggestionApplied: false,
    status: null,
  });

  const activeInstructionsListTab = ref('custom');

  async function addInstruction() {
    newInstruction.status = 'loading';
    activeInstructionsListTab.value = 'custom';

    try {
      const instructionResponse =
        await nexusaiAPI.agent_builder.instructions.addInstruction({
          projectUuid: projectUuid.value,
          instruction: newInstruction,
        });

      instructions.data.unshift({
        ...newInstruction,
        status: 'complete',
        id: instructionResponse.id,
      });
      newInstruction.status = null;
      newInstruction.text = '';

      callAlert('success', 'new_instruction.success_alert');
    } catch (error) {
      newInstruction.status = 'error';
      callAlert('error', 'new_instruction.error_alert');
    }
  }

  async function loadInstructions() {
    instructions.status = 'loading';
    try {
      const response = await nexusaiAPI.agent_builder.instructions.list({
        projectUuid: projectUuid.value,
      });

      instructions.data = [...instructions.data, ...response];
      instructions.status = 'complete';
    } catch (error) {
      instructions.status = 'error';
    }
  }
  async function editInstruction(id, text) {
    const instruction = instructions.data.find(
      (instruction) => instruction.id === id,
    );

    if (!instruction) return;

    try {
      instruction.status = 'loading';
      await nexusaiAPI.agent_builder.instructions.edit({
        projectUuid: projectUuid.value,
        id,
        text,
      });
      instruction.text = text;
      instruction.status = 'complete';

      callAlert('success', 'edit_instruction.success_alert');
    } catch (error) {
      instruction.status = 'error';
      callAlert('error', 'edit_instruction.error_alert');
    }

    return { status: instruction.status };
  }

  async function removeInstruction(id) {
    const instruction = instructions.data.find(
      (instruction) => instruction.id === id,
    );
    if (!instruction) return;
    instruction.status = 'loading';

    try {
      await nexusaiAPI.agent_builder.instructions.delete({
        projectUuid: projectUuid.value,
        id,
      });
      instructions.data = instructions.data.filter(
        (instruction) => instruction.id !== id,
      );
      callAlert('default', 'remove_instruction.success_alert');
      return { status: null };
    } catch (error) {
      instruction.status = 'error';
      callAlert('error', 'remove_instruction.error_alert');
    }

    return { status: instruction?.status };
  }

  async function getInstructionSuggestionByAI() {
    instructionSuggestedByAI.status = 'loading';

    try {
      const response =
        await nexusaiAPI.agent_builder.instructions.getSuggestionByAI({
          projectUuid: projectUuid.value,
          instruction: newInstruction.text,
        });

      instructionSuggestedByAI.data = {
        instruction: newInstruction.text,
        ...response,
      };
      instructionSuggestedByAI.suggestionApplied = false;
      instructionSuggestedByAI.status = 'complete';
    } catch (error) {
      instructionSuggestedByAI.status = 'error';
      callAlert(
        'error',
        'new_instruction.validate_instruction_by_ai.error_alert',
      );
    }
  }

  function updateValidateInstructionByAI(value: boolean) {
    validateInstructionByAI.value = value;
    moduleStorage.setItem('validateInstructionByAI', value);
  }

  function resetInstructionSuggestedByAI() {
    instructionSuggestedByAI.data = {
      instruction: '',
      classification: [],
      suggestion: '',
    };
  }

  return {
    instructions,
    newInstruction,
    validateInstructionByAI,
    instructionSuggestedByAI,
    activeInstructionsListTab,
    addInstruction,
    loadInstructions,
    editInstruction,
    removeInstruction,
    getInstructionSuggestionByAI,
    updateValidateInstructionByAI,
    resetInstructionSuggestedByAI,
  };
});
