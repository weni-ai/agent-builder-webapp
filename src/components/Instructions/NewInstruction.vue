<template>
  <section
    class="new-instruction"
    data-testid="new-instruction"
  >
    <header
      class="new-instruction__header"
      data-testid="new-instruction-header"
    >
      <h2
        class="new-instruction__title"
        data-testid="new-instruction-title"
      >
        {{ $t('agent_builder.instructions.new_instruction.title') }}
      </h2>
      <UnnnicSwitch
        data-testid="new-instruction-switch-validate-instruction-by-ai"
        :modelValue="instructionsStore.validateInstructionByAI"
        :textRight="
          $t(
            'agent_builder.instructions.new_instruction.validate_instruction_by_ai.switch',
          )
        "
        @update:model-value="
          instructionsStore.updateValidateInstructionByAI($event)
        "
      />
    </header>
    <UnnnicTextArea
      v-model="instructionsStore.newInstruction.text"
      data-testid="new-instruction-textarea"
      :placeholder="
        $t('agent_builder.instructions.new_instruction.textarea.placeholder')
      "
      :message="
        $t('agent_builder.instructions.new_instruction.textarea.description')
      "
    />
    <UnnnicButton
      class="new-instruction__add-instruction-button"
      data-testid="add-instruction-button"
      :disabled="!newInstruction.text.trim()"
      :text="primaryButtonText"
      :loading="newInstruction.status === 'loading'"
      @click="handlePrimaryButton"
    />

    <ModalValidateInstruction
      v-if="showValidateInstructionByAIModal"
      v-model="showValidateInstructionByAIModal"
      data-testid="modal-validate-instruction-by-ai"
    />
  </section>
</template>

<script setup>
import { computed, ref } from 'vue';

import i18n from '@/utils/plugins/i18n';

import { useInstructionsStore } from '@/store/Instructions';

import ModalValidateInstruction from './ModalValidateInstruction/index.vue';

const instructionsStore = useInstructionsStore();

const showValidateInstructionByAIModal = ref(false);

const newInstruction = computed(() => instructionsStore.newInstruction);
const primaryButtonText = computed(() => {
  const newInstructionText = (value) =>
    i18n.global.t(`agent_builder.instructions.new_instruction.${value}`);

  return instructionsStore.validateInstructionByAI
    ? newInstructionText('validate_instruction_by_ai.button')
    : newInstructionText('publish_instruction');
});

function openValidateInstructionByAIModal() {
  showValidateInstructionByAIModal.value = true;
}

function handlePrimaryButton() {
  return instructionsStore.validateInstructionByAI
    ? openValidateInstructionByAIModal()
    : instructionsStore.addInstruction();
}
</script>

<style lang="scss" scoped>
.new-instruction {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  padding: $unnnic-spacing-sm;

  display: grid;
  grid-template-columns: repeat(12, 1fr);
  row-gap: $unnnic-spacing-sm;

  :deep(textarea) {
    resize: none;
  }

  & > * {
    grid-column: 1 / -1;
  }

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: $unnnic-space-2;
  }

  &__title {
    margin: 0;

    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__add-instruction-button {
    grid-column: 1 / 5;
  }
}
</style>
