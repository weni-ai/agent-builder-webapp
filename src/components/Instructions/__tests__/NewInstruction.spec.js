import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach, vi } from 'vitest';

import NewInstruction from '../NewInstruction.vue';
import i18n from '@/utils/plugins/i18n';
import { createTestingPinia } from '@pinia/testing';
import { useInstructionsStore } from '@/store/Instructions';

describe('NewInstruction.vue', () => {
  let wrapper;
  let instructionsStore;

  const pinia = createTestingPinia({
    initialState: {
      Instructions: {
        instructions: {
          data: [],
        },
        newInstruction: {
          text: '',
          status: null,
        },
        validateInstructionByAI: true,
      },
    },
  });

  beforeEach(() => {
    vi.clearAllMocks();

    instructionsStore = useInstructionsStore(pinia);

    wrapper = shallowMount(NewInstruction, {
      global: {
        plugins: [pinia],
      },
    });
  });

  const findComponent = (component) =>
    wrapper.findComponent(SELECTORS[component]);
  const find = (selector) => wrapper.find(SELECTORS[selector]);

  const translation = (key) =>
    i18n.global.t(`agent_builder.instructions.new_instruction.${key}`);

  const SELECTORS = {
    container: '[data-testid="new-instruction"]',
    title: '[data-testid="new-instruction-title"]',
    header: '[data-testid="new-instruction-header"]',
    switchValidateInstructionByAI:
      '[data-testid="new-instruction-switch-validate-instruction-by-ai"]',
    textarea: '[data-testid="new-instruction-textarea"]',
    addButton: '[data-testid="add-instruction-button"]',
    modalValidateInstruction:
      '[data-testid="modal-validate-instruction-by-ai"]',
  };

  describe('Component rendering', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders a title, description, textarea and add instruction button', () => {
      expect(find('container').exists()).toBe(true);
      expect(find('title').exists()).toBe(true);
      expect(find('header').exists()).toBe(true);
      expect(findComponent('switchValidateInstructionByAI').exists()).toBe(
        true,
      );
      expect(findComponent('textarea').exists()).toBe(true);
      expect(findComponent('addButton').exists()).toBe(true);
      expect(findComponent('modalValidateInstruction').exists()).toBe(false);
    });

    describe('Title rendering', () => {
      it('renders the correct title', () => {
        expect(find('title').text()).toBe(translation('title'));
      });
    });

    describe('Switch rendering', () => {
      it('renders the correct switch', () => {
        expect(
          findComponent('switchValidateInstructionByAI').props('modelValue'),
        ).toBe(true);
        expect(
          findComponent('switchValidateInstructionByAI').props('textRight'),
        ).toBe(translation('validate_instruction_by_ai.switch'));
      });
    });

    describe('Textarea rendering', () => {
      it('provides the correct props to textarea', () => {
        expect(findComponent('textarea').props('placeholder')).toBe(
          translation('textarea.placeholder'),
        );
        expect(findComponent('textarea').props('message')).toBe(
          translation('textarea.description'),
        );
      });
    });

    describe('Add button rendering', () => {
      it('disables the button when the textarea is empty', () => {
        expect(findComponent('addButton').props('disabled')).toBe(true);
      });
    });

    describe('User interaction', () => {
      it('switches the validate instruction by AI state when the switch is toggled', async () => {
        const switchComponent = findComponent('switchValidateInstructionByAI');
        expect(switchComponent.props('modelValue')).toBe(true);
        await switchComponent.vm.$emit('update:modelValue', false);
        expect(
          instructionsStore.updateValidateInstructionByAI,
        ).toHaveBeenCalledWith(false);
      });

      it('adds an instruction when the primary button is clicked when validate instruction by AI is disabled', async () => {
        instructionsStore.validateInstructionByAI = false;
        await findComponent('addButton').vm.$emit('click');
        expect(instructionsStore.addInstruction).toHaveBeenCalledWith();
      });

      it('does not add instruction when the primary button is clicked when validate instruction by AI is enabled', async () => {
        instructionsStore.validateInstructionByAI = true;
        await findComponent('addButton').vm.$emit('click');
        expect(instructionsStore.addInstruction).not.toHaveBeenCalled();
      });

      it('opens the modal validate instruction when the primary button is clicked when validate instruction by AI is enabled', async () => {
        expect(findComponent('modalValidateInstruction').exists()).toBe(false);
        instructionsStore.validateInstructionByAI = true;
        await findComponent('addButton').vm.$emit('click');
        expect(findComponent('modalValidateInstruction').exists()).toBe(true);
      });
    });
  });
});
