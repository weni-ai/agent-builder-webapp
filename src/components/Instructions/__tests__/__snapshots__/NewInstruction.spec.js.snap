// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewInstruction.vue > Component rendering > should match snapshot 1`] = `
"<section data-v-0b69887d="" class="new-instruction" data-testid="new-instruction">
  <header data-v-0b69887d="" class="new-instruction__header" data-testid="new-instruction-header">
    <h2 data-v-0b69887d="" class="new-instruction__title" data-testid="new-instruction-title">New custom instruction</h2>
    <unnnic-switch-stub data-v-0b69887d="" size="medium" textleft="" textright="Validate instruction by AI" disabled="false" modelvalue="true" usevmodel="false" data-testid="new-instruction-switch-validate-instruction-by-ai"></unnnic-switch-stub>
  </header>
  <anonymous-stub data-v-0b69887d="" modelvalue="" placeholder="You're funny, but don't make jokes" size="md" message="Keep instructions clear, direct, and in the imperative." disabled="false" type="normal" errors="" data-testid="new-instruction-textarea"></anonymous-stub>
  <unnnic-button-stub data-v-0b69887d="" size="large" text="Validate instruction" type="primary" float="false" iconleft="" iconright="" iconcenter="" iconsfilled="false" next="false" disabled="true" loading="false" class="new-instruction__add-instruction-button" data-testid="add-instruction-button"></unnnic-button-stub>
  <!--v-if-->
</section>"
`;
